### PR TITLE
feat: add string compression to query params getter and setter

### PR DESCRIPTION
**Pull request by puskuruk**
Implements #525

### CODE REVIEW

File: packages/fiddle/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "@material-ui/core": "^4.12.4",
     "@material-ui/lab": "^4.0.0-alpha.56",
     "@monaco-editor/react": "^4.4.5",
+    "lz-string": "^1.4.4",
```
[the remark this answers is not in the document]
Comment (puskuruk) [reply packages/fiddle/package.json:15]: Updated 🍻

**Review by samijaber — APPROVED** [review]
https://mitosis.builder.io/?outputTab=vue&code=import+%7B+useStore+%7D+from+%22%40builder.io%2Fmitosis%22%3B%0A%0Aexport+default+function+MyComponent%28props%29+%7B%0A++const+state+%3D+useStore%28%7B%0A++++name%3A+%22Steve%22%0A++%7D%29%3B%0A%0A%0A++return+%28%0A++++%3Cdiv%0A++++++css%3D%7B%7B%0A++++++++padding%3A+%2210px%22%2C%0A++++++%7D%7D%0A++++%3E%0A++++++%3Clink%0A++++++++href%3D%22https%3A%2F%2Funpkg.com%2Ftailwindcss%40%5E2%2Fdist%2Ftailwind.min.css%22%0A++++++++rel%3D%22stylesheet%22%0A++++++%2F%3E%0A%0A++++++%3Cinput%0A++++++++class%3D%22shadow-md+rounded+w-full+px-4+py-2%22%0A++++++++placeholder%3D%22What+is+your+name%3F%22%0A++++++++value%3D%7Bstate.name%7D%0A++++++++onChange%3D%7B%28event%29+%3D%3E+%7B+state.name+%3D+event.target.value+%7D%7D%0A++++++%2F%3E%0A%0A++++++%3Ch1%0A++++++++class%3D%22text-lg%22%0A++++++++css%3D%7B%7B%0A++++++++++marginTop%3A+%2210px%22%2C%0A++++++++%7D%7D%0A++++++%3E%0A++++++++Hello%2C%0A++++++++%7Bstate.name%7D%21%0A++++++%3C%2Fh1%3E%0A++++%3C%2Fdiv%3E%0A++%29%3B%0A%7D

becomes:

https://mitosis-fiddle-git-fork-puskuruk-add-string-c-44eadb-builder-io.vercel.app/?outputTab=G4VwpkA%3D&code=JYWwDg9gTgLgBAbzgVwM4FMDKNrrgXzgDMoIQ4AiAAQCNlgAbAE3SgDpgIB6EYHVYKgoBuAFCj0AD0iw4LIgENkDeEWQA7AMYxO6uAFkAngGEykdenUwAFGFJhUASkSi4cTRHWp43hTDwAvCgY2LjWCK5ucOoKIOgAXJTY6ABu6BSR%2BI5i4m5Q6DDIUHrWkW4APEzAKWVRmqioAQgRUa1wYApMVeoA5okUAIwADGCSFAA0tW74%2BLUAfFNw5QzA6gDWi24AFvlEARRbMDAO8VxcGmBrPWweIFwwCowA7qtM9ahUAHoATFxV3vdHgwXuomGxeOobg0Mm0ovkGPtvIYGOhUFt0AUYW0uAtFuVVmBkDBNu4GAoGoitp0IE8ALQgJhwUgaFiMulqBgMdqSWkAFnahlp3yxsLAZM06C2EGYrH2AHUqfBBHBDBAitFYugAPwitopBQMZDoJq%2BfxsGJxWawtyeYxU3rGhDWVKWGDOAJzRBwU3oc2auBBF1WNgPKA9ApsfWGvAzRY43JtcpbAYkzRkikUfySGC0hg9XWtd5NFrWtwgBRh1YAFQgYH6w1GExJsdhC1LAAl0JyIJNSwgfX7LQBCPFcZNtqLlP7VCfZUT4IA

Definitely an improvement! Thank you for your contribution 😄

**Comment by puskuruk** [comment]
I don't have write access. Can you merge it? @samijaber 💃

**Comment by puskuruk** [comment]
Hmm but the only problem is old links won't work with this change. 😄

**Comment by samijaber** [comment]
Yup! Was waiting on tests to pass. Thank you for your first contribution to the project 😄

**Comment by steve8708** [comment]
Would be nice to get an update so old links work too - we have those sitting around in github issues to reproduce issues so would be nice to not lose that

**Comment by samijaber** [comment]
> Would be nice to get an update so old links work too - we have those sitting around in github issues to reproduce issues so would be nice to not lose that

Very good point, I forgot about those. @puskuruk Would you mind making another PR that handles the old approach too? Perhaps something like:

```ts
export const getQueryParam = (name: string): string | null => {
  const urlParams = new URLSearchParams(window.location.search);
  const compressedValue = urlParams.get(name);

  const decompressedValue = decompressFromBase64(compressedValue);

  return decompressedValue ?? compressedValue;
};
```

I think that should work.

**Comment by puskuruk** [comment]
Sure! It makes sense 💃 I'll test if the implementation above works 🙏

**Comment by puskuruk** [comment]
I've created a [new pr](https://github.com/BuilderIO/mitosis/pull/579) for the issue above.

**Pull request merged**